src/Backends/InterfaceSettings.vala

```diff
@@ -57,5 +81,47 @@ public class SettingsDaemon.Backends.InterfaceSettings : GLib.Object {
         accounts_service.cursor_size = interface_settings.get_int (CURSOR_SIZE);
         accounts_service.locate_pointer = interface_settings.get_boolean (LOCATE_POINTER);
         accounts_service.text_scaling_factor = interface_settings.get_double (TEXT_SCALING_FACTOR);
+
+        accounts_service.picture_options = background_settings.get_enum (PICTURE_OPTIONS);
+        accounts_service.primary_color = background_settings.get_string (PRIMARY_COLOR);
+    }
+
+    private void sync_background_to_greeter () {
+        var source = File.new_for_uri (background_settings.get_string (PICTURE_URI));
+        var source_dark = File.new_for_uri (background_settings.get_string (PICTURE_URI_DARK));
+
+        if (!FileUtils.test (source.get_path (), EXISTS)) {
+            debug ("Wallpaper path is invalid");
+            display_manager_accounts_service.background_file = "";
+            return;
+        }
```

Comment:
you don't need to use `FileUtils.test()` when dealing with GLib.File, `GLib.File.query_exitsts()` will have the same result.